### PR TITLE
chore: Add use cases to developer guide

### LINKED ISSUE #14: Add Use Cases to the Developer Guide
Add Use Cases to the Developer Guide

### PR DESCRIPTION
**Pull request by CloudHill**
Added the following use cases to the developer guide:

* `UC C1` List Customers
* `UC C2` View Customer
* `UC C3` Delete Customer
* `UC C4` Update Customer
* `UC C5` Bookmark or Unbookmark Customer
* `UC C6` Assign Individual to Enterprise
* `UC C7` Create Customer Note
* `UC C8` Append Customer Note ➕
* `UC C9` Delete Customer Note
* `UC O1` List Orders
* `UC O2` Create Order ➕
* `UC O3` View Order
* `UC O4` Delete Order
* `UC O5` Update Order
* `UC O6` Advance Order Status
* ~~`UC O6` Tag Order to Customer~~
* `UC R1` List Rewards
* `UC R2` View Reward
* `UC R3` Delete Reward
* `UC R4` Update Reward
* `UC R5` Set/Add Customer Points ➕
* `UC R6` Give Reward to Customer

Closes #14

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -283,32 +283,391 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-**Use case: Delete a person**
+#### Use case: UC C1 - List Customers
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests list of customers
```
Comment (JavonTeo) [inline docs/DeveloperGuide.md]: A full stop could be used to finish the sentence.

**Review by JavonTeo — CHANGES_REQUESTED** [review]
LGTM. Just a nit to pick: following the use case example in the CS2103T textbook, every line had a full stop, though full stops were not used much here. Other than that, everything looks good!

**Comment by CloudHill** [comment]
Added the full stops, also updated the list of use cases:

* `+ UC C8` Append Customer Note
* `+ UC O2` Create Order
* `- UC O6` Tag Order to Customer
* `+ UC R5` Set/Add Customer Points

**Review by JavonTeo — APPROVED** [review]
LGTM

**Pull request merged**